q2_moshpit/prodigal/prodigal.py

```diff
@@ -24,34 +26,56 @@ def predict_genes_prodigal(
     genes = GenesDirectoryFormat()
     proteins = ProteinsDirectoryFormat()
 
-    # Get paths to fasta files in input dir
-    fasta_files = [
-        file for file in os.listdir(mags.path)
-        if file.endswith(".fa") or file.endswith(".fasta")
-    ]
-
     # Define base command
     base_cmd = [
         "prodigal",
         "-g", translation_table_number,
         "-f", "gff"
     ]
 
-    # For every fasta file in mags.path call prodigal and write
-    # outputs to the corresponding directories.
-    for fasta_file in fasta_files:
-        # Get the filename from the file path
-        file_id = os.path.splitext(fasta_file)[0]
-
-        # Adjust command and run
-        cmd = cp.deepcopy(base_cmd)
-        cmd.extend([
-            "-i", os.path.join(mags.path, fasta_file),
-            "-o", os.path.join(loci.path, f"{file_id}_loci.gff"),
-            "-a", os.path.join(proteins.path, f"{file_id}_proteins.fasta"),
-            "-d", os.path.join(genes.path, f"{file_id}_genes.fasta")
-        ])
-        run_command(cmd)
+    def _process_fasta_files(fasta_files: list, prefix: str, input_path: str):
+        # For every fasta file call prodigal and write
+        # outputs to the corresponding directories.
+        for fasta_file in fasta_files:
+            # Get the filename from the file path
+            file_id = os.path.splitext(fasta_file)[0]
+
+            # Adjust command and run
+            cmd = cp.deepcopy(base_cmd)
+            cmd.extend([
+                "-i", os.path.join(input_path, fasta_file),
+                "-o",
+                os.path.join(
+                    loci.path, f"{prefix}{file_id}_loci.gff"
+                ),
+                "-a",
+                os.path.join(
+                    proteins.path, f"{prefix}{file_id}_proteins.fasta"
+                ),
+                "-d",
+                os.path.join(
+                    genes.path, f"{prefix}{file_id}_genes.fasta"
+                )
+            ])
+            run_command(cmd)
+
+    if isinstance(mags, MAGSequencesDirFmt):
+        # Get paths to fasta files in input dir
+        fasta_files = os.listdir(mags.path)
+        _process_fasta_files(fasta_files, '', mags.path)
```

Comment:
Then here you can have a small loop which will call the prodigal function, which I described above. Also, to not mess with file names you should use the `feacture_dict` method from the `MAGSequencesDirFmt` class which will give you the mapping between mag ID and its path (https://github.com/qiime2/q2-types/blob/5eea61d956135a0d848f15e2fb2910dc72c9c092/q2_types/feature_data_mag/_format.py#L30)